tests/remotesecret/image-pull-remote-secret.go

```diff
@@ -0,0 +1,185 @@
+
+	applicationName := "dotnet-component"
+	gitSourceUrl := "https://github.com/devfile-samples/devfile-sample-dotnet60-basic"
+	environmentName := "image-pull-remote-secret"
+	secret := ""
+
+	AfterEach(framework.ReportFailure(&fw))
+
+	Describe("SVPI-601 - Ensure existence of image pull remote secret and image pull secret when component is created", Ordered, func() {
+		BeforeAll(func() {
+			// Initialize the tests controllers
+			fw, err = framework.NewFramework(utils.GetGeneratedNamespace("spi-demos"))
+			Expect(err).NotTo(HaveOccurred())
+			namespace = fw.UserNamespace
+			Expect(namespace).NotTo(BeEmpty())
+		})
+
+		AfterAll(func() {
+			if !CurrentSpecReport().Failed() {
+				Expect(fw.SandboxController.DeleteUserSignup(fw.UserName)).To(BeTrue())
+			}
+		})
+
+		It("creates an application", func() {
+			createdApplication, err := fw.AsKubeDeveloper.HasController.CreateApplication(applicationName, namespace)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(createdApplication.Spec.DisplayName).To(Equal(applicationName))
+			Expect(createdApplication.Namespace).To(Equal(namespace))
+		})
+
+		It("checks if application is healthy", func() {
+			Eventually(func() string {
+				appstudioApp, err := fw.AsKubeDeveloper.HasController.GetApplication(applicationName, namespace)
+				Expect(err).NotTo(HaveOccurred())
+				application = appstudioApp
+
+				return application.Status.Devfile
+			}, 3*time.Minute, 100*time.Millisecond).Should(Not(BeEmpty()), fmt.Sprintf("timed out waiting for gitOps repository to be created for the %s application in %s namespace", applicationName, fw.UserNamespace))
+
+			Eventually(func() bool {
+				gitOpsRepository := utils.ObtainGitOpsRepositoryName(application.Status.Devfile)
+
+				return fw.AsKubeDeveloper.CommonController.Github.CheckIfRepositoryExist(gitOpsRepository)
+			}, 1*time.Minute, 1*time.Second).Should(BeTrue(), fmt.Sprintf("timed out waiting for HAS controller to create gitops repository for the %s application in %s namespace", applicationName, fw.UserNamespace))
+		})
+
+		It("creates an environment", func() {
+			env, err = fw.AsKubeDeveloper.GitOpsController.CreatePocEnvironment(environmentName, namespace)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("creates component detection query", func() {
+			cdq, err = fw.AsKubeDeveloper.HasController.CreateComponentDetectionQuery(applicationName, namespace, gitSourceUrl, "", "", secret, false)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("creates components", func() {
+			for _, compDetected := range cdq.Status.ComponentDetected {
+				c, err := fw.AsKubeDeveloper.HasController.CreateComponent(compDetected.ComponentStub, namespace, "", secret, applicationName, true, map[string]string{})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(c.Name).To(Equal(compDetected.ComponentStub.ComponentName))
+
+				componentList = append(componentList, c)
+			}
+		})
+
+		It("waits for components pipeline to be finished", func() {
```

Comment:
I think we don't need to check for these steps  `pipeline to be finished`, `snapshot to be marked as successful`, `SnapshotEnvironmentBinding is created successfully`, `deploys components of application successfully`, since we are already do these verification in rhtap-demo tests, doing it here will be a duplicate effort. After creating component, we can directly check `if image pull remote secret was deployed and if image pull secret exists`. WDYT?

Reply:
@tisutisu, I thought about it but I was afraid that if something goes wrong while creating/deploying the component it will be difficult to debug without those steps. WDYT?

Reply:
Suppose image pull remote secret functionality is broken, then I think this test will fail at step while checking`build pipeline run to be successful` in that case we should put the remote secret check before checking `pipeline run to be successful`. @mmorhun Does the image pull remote secret has anything to do with other steps where we are verifying `SnapShot`,  `SnapshotEnvironmentBinding`, `Deployment` ?

Reply:
To my mind, we should check only related to Remote Secret functionality here. In my understanding this includes the following checks:
- Remote Push Secret is created and has correct settings (secret type, target, labels, annotations) and healthy (its status reports state injected and so on).
- Push secret is created and linked to `appstudio-pipeline` service account.
- Remote Pull Secret is created and has correct settings and status. Targets are set according to the annotations/labels.
- Pull secrets are propagated to target environments and linked to the default service account.
- Optionally check that the tokens in the resulted push/pull secrets are correct.
Waiting for the pipeline to finish, checking snapshots and deployments is out of scope in my opinion.

Reply:
This PR will only address image pull remote secret, for now. The next one will address also the image push remote secret

Reply:
I will only remove `Deployment` verification since the other verifications are needed. Targets appear in RemoteSecret when SnapshotEnvironmentBinding is updated after the build pipeline finishes.

Reply:
I will only remove `Deployment` verification since the other verifications are needed. Targets appear in RemoteSecret when SnapshotEnvironmentBinding is updated after the build pipeline finishes.